projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,229 @@
+		<h1 class="wp-block-heading has-text-align-center">$site_name</h1>
+		<!-- /wp:heading -->
+
+		<!-- wp:paragraph {"align":"center"} -->
+		<p class='has-text-align-center'>$site_description</p>
+		<!-- /wp:paragraph -->
+
+		<!-- wp:jetpack/subscriptions /-->
+
+		<!-- wp:paragraph {"align":"center","className":"jetpack-subscribe-overlay__to-content"} -->
+		<p class="has-text-align-center jetpack-subscribe-overlay__to-content"><a href="#">$skip_to_content</a> ↓</p>
+		<!-- /wp:paragraph -->
+	</div>
+	<!-- /wp:group -->
+HTML;
+	}
+
+	/**
+	 * Enqueues JS to load overlay.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets() {
+		if ( $this->should_user_see_overlay() ) {
+			wp_enqueue_style( 'subscribe-overlay-css', plugins_url( 'subscribe-overlay.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-overlay-js', plugins_url( 'subscribe-overlay.js', __FILE__ ), array( 'wp-dom-ready' ), JETPACK__VERSION, true );
+		}
+	}
+
+	/**
+	 * Adds overlay with Subscribe Overlay content.
+	 *
+	 * @return void
+	 */
+	public function add_subscribe_overlay_to_frontend() {
+		if ( $this->should_user_see_overlay() ) { ?>
+					<div class="jetpack-subscribe-overlay">
+						<div class="jetpack-subscribe-overlay__close">
+							<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+								<path d="M5.40456 5L19 19M5 19L18.5954 5" stroke="currentColor" stroke-width="1.5"/>
+							</svg>
+						</div>
+						<div class="jetpack-subscribe-overlay__content">
+							<?php block_template_part( self::BLOCK_TEMPLATE_PART_SLUG ); ?>
+						</div>
+					</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Returns true if a site visitor should see
+	 * the Subscribe Overlay.
+	 *
+	 * @return bool
+	 */
+	public function should_user_see_overlay() {
+		// Only show when viewing frontend.
+		if ( is_admin() ) {
+			return false;
+		}
+
+		// Needed because Elementor editor makes is_admin() return false
+		// See https://coreysalzano.com/wordpress/why-elementor-disobeys-is_admin/
+		// Ignore nonce warning as just checking if is set
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['elementor-preview'] ) ) {
+			return false;
+		}
+
+		// Don't show when previewing blog posts or site's theme
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['preview'] ) || isset( $_GET['theme_preview'] ) || isset( $_GET['customize_preview'] ) || isset( $_GET['hide_banners'] ) ) {
+			return false;
+		}
+
+		// Don't show if one of subscribe query params is set.
+		// They are set when user submits the subscribe form.
+		// The nonce is checked elsewhere before redirect back to this page with query params.
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['subscribe'] ) || isset( $_GET['blogsub'] ) ) {
+			return false;
+		}
+
+		// Don't show if post is for subscribers only or has paywall block
+		global $post;
```

Comment:
Let's limit the welcome modal to site's homepage, and blog page if blog page is different from homepage (WP has functions to detect these).

Posts will have the other smaller modal and other pages can be without the modal.

Reply:
Sure, `is_home() || is_front_page()` should do the trick.